apps/api/v1/pages/api/bookings/[id]/_patch.ts

```diff
@@ -109,14 +110,27 @@ export async function patchHandler(req: NextApiRequest) {
 }
 
 async function checkPermissions(req: NextApiRequest, body: z.infer<typeof schemaBookingEditBodyParams>) {
-  const { isAdmin } = req;
-  if (body.userId && !isAdmin) {
+  const { userId, isSystemWideAdmin, isOrganizationOwnerOrAdmin } = req;
+  if (body.userId && !isSystemWideAdmin && !isOrganizationOwnerOrAdmin) {
     // Organizer has to be a cal user and we can't allow a booking to be transfered to some other cal user's name
     throw new HttpError({
       statusCode: 403,
       message: "Only admin can change the organizer of a booking",
     });
   }
+
+  if (body.userId && isOrganizationOwnerOrAdmin) {
+    const accessibleUsersIds = await getAccessibleUsers({
+      adminUserId: userId,
+      memberUserIds: [body.userId],
```

Comment:
Are we calling this in the endpoint and in the `auth-middleware`? This is starting to make me concerned about the performance for bigger resources.

Reply:
It is generally not done in middleware because not each endpoint would require this, so if we put this in the middleware, it would make queries even when it doesn't need to. 
Two queries are made, one to get member's memberships (for all users provided, one if only one) and admin memberships (always one). I could think of getting both in the same query but I wonder if the performance difference would be this big, as it might be at the expense of clarity and scalability of the code itself. 

Can you elaborate, what is your concern exactly in terms of performance for bigger resources here?

Reply:
What I think I should do here though, is perhaps reduce the number of queries made. I think if I merge the two DB queries and then only perform meaningful data extraction for member and admin memberships, that'll reduce the overhead. 👀

Reply:
> Are we calling this in the endpoint and in the auth-middleware?

We only call this when necessary and it isn't called more than once per request, be it `/bookings POST`, `/bookings GET`. Then, since for the specific booking ID paths, admin checks happen in middleware, it is done only in the `auth-middleware`, and **not** the subsequent endpoints(`/bookings/[id] — PATCH/DELETE/GET`). I hope that clears it a bit